src/OpenApi/src/Extensions/OpenApiEndpointRouteBuilderExtensions.cs

```diff
@@ -43,7 +43,7 @@ public static IEndpointConventionBuilder MapOpenApi(this IEndpointRouteBuilder e
                 }
                 else
                 {
-                    var document = await documentService.GetOpenApiDocumentAsync();
+                    var document = await documentService.GetOpenApiDocumentAsync(context.RequestAborted);
                     var documentOptions = options.Get(documentName);
                     using var output = MemoryBufferWriter.Get();
                     using var writer = Utf8BufferTextWriter.Get(output);
```

Comment:
Not part of this PR, but on line 53 you call `output.ToArray()` which seems wasteful, was `BodyWriter.CopyFromAsync(output)` evaluated?